src/data_structures/DreeNode.h

```diff
@@ -17,6 +17,12 @@ class DreeNode {
     DreeNode *right;
 
     DreeNode(std::string &str);
+
+    ~DreeNode() {
```

Comment:
wont it increase the latency? Latency is already very high due to iterations. Lets not do this part. We will have to find some other way. Maybe concurrently. There is a separate issue for that.

Reply:
Sure will look for other possibilities and discuss